### ISSUE TITLE
Always get unauthorized access when setting up an openid authentication 

### ISSUE THREAD

**Issue by ped-suzanto** [ThreeMammals/Ocelot#1187]
## Expected Behavior
Authentication using openId should work properly

## Actual Behavior
it always returns 401, no matter if the authority exists or not

## Steps to Reproduce the Problem
my ConfigureServices is:
```
services.AddAuthentication(options => {
                options.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
                options.DefaultChallengeScheme = "OrchardOpenId";
            })
                .AddCookie()
                .AddOpenIdConnect("OrchardOpenId", options =>
                {
                    options.Authority = "https://localhost:3000";
                    options.ClientId = "web-api";
                    options.ClientSecret = "38624481a0e34f56bfbb5d9557b38937";
                    options.ResponseType = OpenIdConnectResponseType.Code;
                    options.Scope.Add("openid");
                });
            services.AddOcelot();
```
  1. the authentication is working fine if I'm not using ocelot
  2. I am suspecting the cookie authentication is the source of the unauthorized error

## Specifications

  - Version: 14.x.x
  - Platform: Windows
  - Subsystem:

**Comment by xyfy** [comment]
I got a similar question.

when I pass `bearer` token  on the header.

the ocelot does not validate the token if the url do not need authorize.

I want the ocelot go check it ,when I pass the token.        if I do not pass it ,do nothing is ok

```
[14:58:25 DBG] AuthenticationScheme: Bearer was not authenticated.
[14:58:25 DBG] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: ocelot pipeline started
[14:58:25 DBG] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: Upstream url path is /api/ContentManager/article/all
[14:58:25 DBG] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: downstream templates are /api/ContentManager/{everything}
[14:58:25 INF] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: EndpointRateLimiting is not enabled for /api/ContentManager/{everything}
[14:58:25 INF] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: No authentication needed for /api/ContentManager/article/all
[14:58:25 INF] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: /api/ContentManager/{everything} route does not require user to be authorised
[14:58:26 DBG] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: Downstream url is http://localhost:55565/api/ContentManager/article/all
[14:58:26 DBG] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: setting http response message
[14:58:26 DBG] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: no pipeline errors, setting and returning completed response
[14:58:26 DBG] requestId: 0HLVIAG4VSOF8:00000001, previousRequestId: no previous request id, message: ocelot pipeline finished
```

**Comment by raman-m** [comment]
### We are not a consulting firm!

But I will explain...
You tried to register 2 auth providers but you hadn't shown us JSON config. This is the First mistake!

Your deafult scheme is `CookieAuthenticationDefaults.AuthenticationScheme`. I assume this provider affected your 2nd one.
That is why you had strange behavior.

Finally, I would say you used multiple auth schemes scenario! That was the root cause.
[Multiple Authentication Schemes](https://ocelot.readthedocs.io/en/latest/features/authentication.html#multiple-authentication-schemes) feature was implemented in version [23.0.0](https://github.com/ThreeMammals/Ocelot/releases/tag/23.0.0) only :))))
